### PR TITLE
Feature/ai

### PR DESCRIPTION
**Pull request by eszterhorvath**

### CODE REVIEW

File: ToDo.Core.Test/AddViewModelTest.cs [hunk 1]
```diff
@@ -85,5 +90,53 @@ public async Task WhenAddingATodo_WithoutTitle_ShowsAlertAndDoesNotAddTheTodo()
...
+            await _viewModel.TakePhoto();
+
+            // ASSERT
+            _computerVisionService.Verify(s => s.GetImageSummary(testPhoto), Times.Once());
+            _computerVisionService.Verify(s => s.RecognizeText(testPhoto), Times.Once());
+            _computerVisionService.Verify(s => s.RecognizeTags(testPhoto), Times.Once());
+            _computerVisionService.Verify(s => s.RecognizeObjects(testPhoto), Times.Once());
+            _computerVisionService.Verify(s => s.RecognizeBrands(testPhoto), Times.Once());
+            _computerVisionService.Verify(s => s.RecognizeFaces(testPhoto), Times.Once());
+        }
```
Comment (gentledepp) [inline ToDo.Core.Test/AddViewModelTest.cs]: As the two tests have the same "Arrange" and the same "Act", they could be merged (just combine the Asserts of both).
Having both of them does not make sense

File: ToDo.Core/Services/IComputerVisionService.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public interface IComputerVisionService
+    {
+        Task<string> RecognizeText(MediaFile photo);
+        Task<string> GetImageSummary(MediaFile photo);
+        Task<List<DetectedObject>> RecognizeObjects(MediaFile photo);
+        Task<List<FaceDescription>> RecognizeFaces(MediaFile photo);
+        Task<List<DetectedBrand>> RecognizeBrands(MediaFile photo);
+        Task<List<ImageTag>> RecognizeTags(MediaFile photo);
+    }
```
Comment (gentledepp) [inline ToDo.Core/Services/IComputerVisionService.cs:20]: Well done! ✨

File: ToDo.Core/ViewModels/AddViewModel.cs [hunk 3]
```diff
@@ -18,18 +22,110 @@ public class AddViewModel : MvxViewModel
...
 
         public Models.ToDo ToDoItem { get; set; }
         public ICommand AddTodoItemCommand { get; }
+        public ICommand TakePhotoCommand { get; }
 
-        public AddViewModel(IToDoService todoService, IMvxNavigationService navigationService, IUserDialogs userDialogService)
+        private Stream photoStream;
+
+        public AddViewModel(IToDoService todoService, IMvxNavigationService navigationService, IUserDialogs userDialogService,
+            IComputerVisionService computerVisionService, IMediaService mediaService)
```
Comment (gentledepp) [inline ToDo.Core/ViewModels/AddViewModel.cs:35]: You see now how fast the list of dependencies of a viewmodel adds up.
There are ways, you can get around this though: Using a Mediator pattern:
https://lostechies.com/jimmybogard/2015/05/05/cqrs-with-mediatr-and-automapper/ 
We are thinking about using this approach for our app as well ;-)

**Review by gentledepp — APPROVED** [review]

**Comment by gentledepp** [comment]
Hi Eszter! I approved the PR - do you want to push "Merge"?

**Comment by eszterhorvath** [comment]
I merged the tests.
And yes, thank you! 😍

**Pull request merged**